tests/test_web_sendfile_functional.py

```diff
@@ -802,7 +802,7 @@ def test_static_route_path_existence_check() -> None:
         tr = request.transport
         sock = tr.get_extra_info('socket')
         sock.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 1024)
-        ret = web.FileResponse(pathlib.Path(str(tmpdir.join(filename))))
+        ret = web.FileResponse(pathlib.Path(str((tmp_path / filename))))
```

Comment:
here and in a few other places now the path is turned into string and than back to Path.

Reply:
Sorry, I've missed this.
Please feel free to make a PR for the fix.
The problem is minor though if all lines belong to tests only

Reply:
Good point. I'm working on a PR now to more comprehensively address path handling across the test suite.

Mostly that's replacing `os.path` calls with their appropriate `pathlib.Path` counterparts.

There are also a couple of uses of ad-hoc temporary directories that I'm updating to `tmp_path`.

Reply:
Ironically, to support 3.5 some `pathlib.Path(str(tmp_path)))` turns out to be necessary. 

pytest uses pathlib2 to provided tmp_path for Python 3.5, which is mostly fine but it does  create a couple of corner cases of incompatibility where pathlib2.Path fails isinstance checks (which require str or PurePath). See pytest-dev/pytest/issues/5017